### PR TITLE
more stricter required labels checking,

### PR DESCRIPTION
**Pull request by rcerven**
name and component needs to have explicit value,
version, release may be using env variable

* OSBS-7697

Signed-off-by: Robert Cerven <rcerven@redhat.com>

### CODE REVIEW

File: osbs/api.py [hunk 1]
```diff
@@ -513,6 +513,9 @@ def _check_labels(self, repo_info):
 
         required_missing = False
         req_labels = {}
+        # required labels which needs to have explicit value (not from env variable)
+        strict_labels = [utils.Labels.LABEL_TYPE_NAME,
```
Comment (athos-ribeiro) [inline osbs/api.py]: could we call them explicit_labels?

File: osbs/api.py [hunk 2]
```diff
@@ -521,14 +524,19 @@ def _check_labels(self, repo_info):
                       utils.Labels.LABEL_TYPE_VERSION]:
             try:
                 _, req_labels[label] = labels.get_name_and_value(label)
+
+                if label in strict_labels and not req_labels[label]:
```
Comment (athos-ribeiro) [inline osbs/api.py]: Could this block be in the try/except `else` clause?

File: osbs/api.py [hunk 3]
```diff
@@ -521,14 +524,19 @@ def _check_labels(self, repo_info):
...
+                                 labels.get_name(label))
             except KeyError:
                 required_missing = True
                 logger.error("required label missing from Dockerfile : %s",
                              labels.get_name(label))
 
         try:
             _, release_value = labels.get_name_and_value(utils.Labels.LABEL_TYPE_RELEASE)
-            if not RELEASE_LABEL_FORMAT.match(release_value):
+            if release_value and not RELEASE_LABEL_FORMAT.match(release_value):
```
Comment (athos-ribeiro) [inline osbs/api.py]: Could this block also be moved to the `else` clause?

**Comment by athos-ribeiro** [comment]
Looks good!

Could we rephrase the commit message to something like "Enforce explicit labels" ?

File: osbs/api.py (hunk 1 above)
Comment (rcerven) [reply osbs/api.py]: sure

File: osbs/api.py (hunk 2 above)
Comment (rcerven) [reply osbs/api.py]: I'd rather leave it here as it is more visible than in else after except

File: osbs/api.py (hunk 3 above)
Comment (rcerven) [reply osbs/api.py]: I'd rather leave it here as it is more visible than in else after except

**Comment by rcerven** [comment]
commit message changed

**Comment by rcerven** [comment]
rn updated: - Enforce explicit required labels, name and component needs to have explicit value, version, release may be using env variable

**Pull request merged**